### PR TITLE
Add basic validator for validating renewal and expiry records

### LINKED ISSUE #4846: Make sure it shows that an account balance does not go negative and then we delete an account. It should all show up in the record stream the right way. -- With and without royalties the right stuff should show up in the record stream.
- [x] There shouldn't be royalties and fees when the balance goes back to the treasury.
- [x] We should also show the correct order of operations when tokens are transferred out or account is renewed in the record stream so that it can be displayed properly in mirror node and hashscan.

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #4846 
Adds a basic validator for validating renewal and expiry records.

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/junit/ExpiryRecordsValidator.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+import com.hederahashgraph.api.proto.java.TransactionRecord;
+import java.util.ArrayList;
+import java.util.List;
+
+/** This validator validates expiry contract records at the end of test run */
+public class ExpiryRecordsValidator implements RecordStreamValidator {
+    private static final String AUTO_RENEWAL_MEMO = " was automatically renewed";
+    private static final String AUTO_EXPIRY_MEMO = " was automatically deleted";
+
```
Comment (Neeharika-Sompalli) [inline test-clients/src/main/java/com/hedera/services/bdd/junit/ExpiryRecordsValidator.java:30]: Open to any suggestions on any other validations that  can be added.

Comment (mhess-swl) [reply test-clients/src/main/java/com/hedera/services/bdd/junit/ExpiryRecordsValidator.java:30]: Would it be helpful to validate that the sum total of the transfers is 0? That's the only thing I can think of

**Review by mhess-swl — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by povolev15 — APPROVED** [review]
LGTM

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by mhess-swl — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by iwsimon — DISMISSED** [review]
LGTM

**Review by mhess-swl — APPROVED** [review]

**Review by povolev15 — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]
LTGM

**Pull request merged**